Userland/Libraries/LibGfx/Filters/StackBlurFilter.cpp

```diff
@@ -0,0 +1,329 @@
+struct BlurStack {
+    BlurStack(size_t size)
+    {
+        m_data.resize(size);
+    }
+
+    struct Iterator {
+        friend BlurStack;
+
+        ALWAYS_INLINE Color& operator*()
+        {
+            return m_data.at(m_idx);
+        }
+
+        ALWAYS_INLINE Color* operator->()
+        {
+            return &m_data.at(m_idx);
+        }
+
+        ALWAYS_INLINE Iterator operator++()
+        {
+            // Note: This seemed to profile slightly better than %
+            if (++m_idx >= m_data.size())
+                m_idx = 0;
+            return *this;
+        }
+
+        ALWAYS_INLINE Iterator operator++(int)
+        {
+            auto prev_it = *this;
+            ++*(this);
+            return prev_it;
+        }
+
+    private:
+        Iterator(size_t idx, Span<Color> data)
+            : m_idx(idx)
+            , m_data(data)
+        {
+        }
+
+        size_t m_idx;
+        Span<Color> m_data;
+    };
+
+    Iterator iterator_from_position(size_t position)
+    {
+        VERIFY(position < m_data.size());
+        return Iterator(position, m_data);
+    }
+
+private:
+    Vector<Color, 512> m_data;
+};
+
+// This is an implementation of StackBlur by Mario Klingemann (https://observablehq.com/@jobleonard/mario-klingemans-stackblur)
+// (Link is to a secondary source as the original site is now down)
+FLATTEN void StackBlurFilter::process_rgba(u8 radius)
+{
+    // TODO: Implement a plain RGB version of this (if required)
+
+    using uint = unsigned;
+
+    if (radius == 0)
+        return;
+
+    constexpr auto transparent_white = Color(Color::White).with_alpha(0);
+
+    uint width = m_bitmap.width();
+    uint height = m_bitmap.height();
+
+    uint div = 2 * radius + 1;
+    uint radius_plus_1 = radius + 1;
+    uint sum_factor = radius_plus_1 * (radius_plus_1 + 1) / 2;
+
+    auto get_pixel = [&](int x, int y) {
+        auto color = m_bitmap.get_pixel<StorageFormat::BGRA8888>(x, y);
+        if (color.alpha() == 0)
+            return transparent_white;
+        return color;
+    };
+
+    auto set_pixel = [&](int x, int y, Color color) {
+        return m_bitmap.set_pixel<StorageFormat::BGRA8888>(x, y, color);
+    };
+
+    BlurStack blur_stack { div };
+    auto const stack_start = blur_stack.iterator_from_position(0);
+    auto const stack_end = blur_stack.iterator_from_position(radius_plus_1);
+    auto stack_iterator = stack_start;
+
+    // Note: `(value * sum_mult[radius]) >> shift_table[radius]` closely approximates a division by the radius squared.
+    auto const sum_mult = mult_table[radius];
+    auto const sum_shift = shift_table[radius];
+
+    for (uint y = 0; y < height; y++) {
+        stack_iterator = stack_start;
+
+        auto color = get_pixel(0, y);
+        for (uint i = 0; i < radius_plus_1; i++)
+            *(stack_iterator++) = color;
+
+        // All the sums here work to approximate a gaussian.
+        uint red_in_sum = 0;
```

Comment:
Not mandatory:
These could easily use some SIMD types, to possibly paralleize the multiplies and additions
It would also condense the code/avoid repetition of nearly the same code

Reply:
I'll give that a go later today and see how that effects the performance

Reply:
...had enough other issues crop up that I don't want to look at SIMD now too. This yak is free for the taking.